### PR TITLE
ConfigFactorySelector

### PR DESCRIPTION
**Pull request by kgromadzki**

### CODE REVIEW

File: whylog/config/__init__.py [hunk 1]
```diff
@@ -109,12 +109,24 @@ def load_config(cls, path):
...
             config_file.write(yaml.safe_dump(config_paths, explicit_start=True))
         return path_to_config
 
 
+class ConfigFactorySelector(object):
+    SUPPORTED_TYPES = {'yaml': YamlConfigFactory}
+
+    @classmethod
+    def get_config(cls):
+        return cls.SUPPORTED_TYPES['yaml'].get_config()
```
Comment (andrzejgorski) [inline whylog/config/__init__.py]: This function should find .whylog/config.yaml file and load the type of config (in this case it should be 'yaml') and then call get_config method with param - rest of the content of loaded file with specific paths and others initial params stored in the dict. Yaml files supports the dict structure.

File: whylog/config/__init__.py [hunk 2]
```diff
@@ -109,12 +109,24 @@ def load_config(cls, path):
...
     @classmethod
-    def create_file_with_config_paths(cls, config_paths, path_to_config):
+    def _create_file_with_config_paths(cls, config_paths, path_to_config):
         with open(path_to_config, 'w') as config_file:
             config_file.write(yaml.safe_dump(config_paths, explicit_start=True))
         return path_to_config
 
 
+class ConfigFactorySelector(object):
+    SUPPORTED_TYPES = {'yaml': YamlConfigFactory}
```
Comment (ppolewicz) [inline whylog/config/__init__.py]: Order of configs must be preserved. You will need to decide which one of many existing configs to load.

File: whylog/tests/tests_config/__init__.py [hunk 3]
```diff
@@ -107,7 +107,7 @@ def setUpClass(cls):
 
         path_config = ['whylog', 'tests', 'tests_config', 'test_files', '.whylog', 'config.yaml']
         path = os.path.join(*path_config)
-        cls.config, _ = YamlConfigFactory.load_config(path)
+        cls.config, _ = ConfigFactorySelector.load_config(path)
```
Comment (ppolewicz) [inline whylog/tests/tests_config/__init__.py]: This is how it should look like :+1:

File: whylog/config/__init__.py [hunk 4]
```diff
@@ -339,3 +240,117 @@ def get_rules_for_log_type(self, log_type):
...
+    }
+
+    @classmethod
+    def _create_settings_dict(cls, whylog_dir):
+        settings = {}
+        for key, file_name in six.iteritems(cls.FILES_NAMES):
+            path = os.path.join(whylog_dir, file_name)
+            cls._create_empty_file(path)
+            settings[key] = path
+        settings['pattern_assistant'] = 'regex'
```
Comment (ppolewicz) [inline whylog/config/__init__.py]: Pull the default values up to AbstractSettingsFactory

**Pull request merged**